### PR TITLE
fix: broken Yarn PnP binary finder strategy

### PR DESCRIPTION
**Pull request by Wroud**
### Summary

Fix resolving binaries with Yarn PnP mode

### Checklist

- [ ] I have tested my changes on the following platforms:
  - [ ] Windows
  - [ ] Linux
  - [x] macOS

### CODE REVIEW

File: test/fixtures/yarn-pnp/biome.json
Comment (Wroud) [inline test/fixtures/yarn-pnp/biome.json:1]: this file needed to allow initializing yarn inside pnpm project

File: lefthook.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ pre-commit:
   commands:
     check:
       glob: "*.{js,ts,json}"
-      run: biome check --apply {staged_files} && git add {staged_files}
```
Comment (ematipico) [inline lefthook.yaml:5]: Do you know if this change was intended? It doesn't seem to be part of the PR

File: test/fixtures/yarn-pnp/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Yarn PnP
+
+This folder contains an example single-root workspace used as a test fixture for
+the Biome extension. The workspace folder is setup as a yarn pnp project.
```
Comment (ematipico) [inline test/fixtures/yarn-pnp/README.md:4]: If this is an example folder, wouldn't having these files in an `examples/` folder be more coherent?

File: lefthook.yaml (hunk 1 above)
Comment (Wroud) [reply lefthook.yaml:5]: In my case, I cloned this repo for the first time, and I haven't had a globally installed biome before. Because of that, this validation failed when I tried to commit.

This change uses the local biome version installed with pnpm instead of the global version, which is not guaranteed to be installed on the machine.

File: test/fixtures/yarn-pnp/README.md (hunk 2 above)
Comment (Wroud) [reply test/fixtures/yarn-pnp/README.md:4]: I've used the same template as other fixtures for consistency. I can definitely change it, but then it would be inconsistent. What do you think?

File: lefthook.yaml (hunk 1 above)
Comment (ematipico) [reply lefthook.yaml:5]: Yeah, that makes sense!
